packages/sdk-js/src/feature-repository.ts

```diff
@@ -277,18 +369,21 @@ async function fetchFeatures(
             clientKey,
             error: e ? e.message : null,
           });
-        activeFetches.delete(key);
+        activeFetches.delete(cacheKey);
         return Promise.resolve({});
       });
-    activeFetches.set(key, promise);
+    activeFetches.set(cacheKey, promise);
   }
   return await promise;
 }
 
 // Watch a feature endpoint for changes
 // Will prefer SSE if enabled, otherwise fall back to cron
 function startAutoRefresh(instance: GrowthBook): void {
-  const [key, apiHost, clientKey] = getKey(instance);
+  const key = getKey(instance);
+  const cacheKey = getCacheKey(instance);
+  const { streamingHost, apiRequestHeaders } = instance.getApiHosts();
```

Comment:
Separate `streamingRequestHeaders` setting instead of reusing `apiRequestHeaders`